src/main/java/teammates/common/datatransfer/InstructorAttributes.java

```diff
@@ -324,18 +314,9 @@ public boolean hasTutorPrivileges() {
         return privileges.hasTutorPrivileges();
     }
 
-    /**
-     * pre-condition: instructorPrivilegesAsText and privileges should be non-null
-     * @param instructor
-     * @return
-     */
     public boolean isEqualToAnotherInstructor(InstructorAttributes instructor) {
-        if (gson.toJson(this).equals(gson.toJson(instructor))) {
-            return true;
-        } else {
-            return !instructorPrivilegesAsText.equals(instructor.instructorPrivilegesAsText) 
-                   && privileges.equals(instructor.privileges);
-        }
+        JsonParser parser = new JsonParser();
+        return parser.parse(getJsonString()).equals(parser.parse(instructor.getJsonString()));
```

Comment:
The json strings obtained from 2 equal `InstructorAttributes` may not the same because the order of the key value pairs in the json strings may be different. Therefore we cannot compare the json strings directly, instead we need to parse them into json objects before comparison.